### ISSUE TITLE
Assign Okta App to Sign On > User authentication > Authentication Policy

### ISSUE THREAD

**Issue by flyweight** [okta/terraform-provider-okta#1086]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There is currently an API call to reassign an [Application to an Authentication Policy](https://developer.okta.com/docs/reference/api/apps/#update-application-policy): `PUT /api/v1/apps/{appId}/policies/{policyId}` but I cannot figure out how to reassign an Application to a different Authentication Policy in the Okta Terraform Provider.  There seems to be a Data Source for fetching the currently applied Policy for an application (okta_app_signon_policy), and Resources for creating a new policy & modifying policy rules (okta_app_signon_policy_rule, okta_policy_rule_signon, okta_policy_signon).

Am I missing how I can assign my apps to an Authentication Policy?

### New or Affected Resource(s)

- okta_app_signon_policy

### Potential Terraform Configuration

```hcl

# If implemented this way, it might be nice to have a data resource that can fetch sign on policies by name (e.g. "Default Policy", "Okta Browser Plugin", "Password only", etc)

resource "okta_app_oauth" "example" {
 ...
}

resource "okta_app_signon_policy" "example" {
  app_id = okta_app_oauth.example.id
}

# OR

resource "okta_app_oauth" "example" {
 signon_policy_id = "123"
}
```

**Comment by monde** [comment]
Thanks for the explanation and API links @flyweight . I will have to look into this.

**Comment by flyweight** [comment]
Thanks @monde. Currently when an application is created, it is assigned to the "Default Policy". The `okta_app_signon_policy_rule` can be used to create another rule inside the "Default Policy" which is first-match, priority-based. Let me know if I missed something, or if you need more info.

**Comment by jglopez** [comment]
Hey @flyweight, I'm running into this too.
I believe that since Authentication Policies were previously called "App Sign-on Policies" and could only be associated with a single application, the provider has not yet been updated to now include the ability to create a new policy and define it within `okta_app_oauth`, `okta_app_saml`, etc.
I agree that all the current documentation describes is how to define (app) authentication policy rules only for the "current" app by using the `okta_app_signon_policy` data source to "link" the app and policy together like you described.

https://support.okta.com/help/s/article/Authentication-Policy-Sharing-Overview
https://developer.okta.com/docs/release-notes/2022-okta-identity-engine/#shareable-authentication-policies

**Comment by Mrorya** [comment]
+1 to the previous comment - would love to make use of the linked authentication policy sharing as an easier way to manage these updates in terraform without having to create a unique policy resource per application

The API documentation stating `PUT to /api/v1/apps/{appId}/policies/{policyId}.` as a method to add a policyId to multiple appIds suggests in the future we'd ideally be able to add multiple app_id references to an existing policy in tf - would love to see this enhancement @monde !

**Comment by jglopez** [comment]
Just noticed https://github.com/okta/terraform-provider-okta/pull/1193
Is this fixed with [v3.31.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.31.0)? 😮

**Comment by flyweight** [comment]
> Just noticed #1193 Is this fixed with [v3.31.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.31.0)? 😮

Yes, thanks for the help.

https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_signon_policy#example-usage

For reference, the resource argument `authentication_policy` for `okta_app_oauth` and `okta_app_saml` is now available to link the app to a signon policy.
